src/main/java/org/opentripplanner/model/BookingRule.java

```diff
@@ -0,0 +1,182 @@
+/* This file is based on code copied from project OneBusAway, see the LICENSE file for further information. */
+package org.opentripplanner.model;
+
+public final class BookingRule {
+
+    private FeedScopedId id;
```

Comment:
This should be final

Reply:
I believe `super(id)` in the parent class does the trick. 🤔 At least that is the way `FareAttribute` class is set.